### DISCUSSION TITLE
Advances settings

### DISCUSSION THREAD

**Discussion by YorikHansen** [jxn-30/better-moodle#357] [Ideas]
With more [incredibly useful features](https://github.com/jxn-30/better-moodle/commit/1730258347f2b0a3d2b03745d11f55468905de64) to be added to Better-Moodle and Better-Moodles philosophy to have pretty much everything customizable (or at least most features toggleable)... 

<details><summary>...the settings menu gets longer and longer.</summary>

![Better-Moodle_settings](https://github.com/user-attachments/assets/159ce850-436c-4e47-a573-80a55b5f09fd)

</details>

I would categorize some of these settings as "advanced", as many settings might not be relevant for the average user.

**Comment by jxn-30** [comment]
What if we used [Bootstrap Tabs](https://getbootstrap.com/docs/4.6/components/navs/#tabs) or [Bootstrap Collapse](https://getbootstrap.com/docs/4.6/components/collapse/)? One could do this to create some kind of "stepper" form which groups the Settings-Groups into several pages. Those could be similiar to:
* basic
* advanced
* fun

Let's put this onto the ideas list of v2 so that we consider it when architecturing settings.

**Reply by YorikHansen** [reply]
Adding this to v2 is a great idea. 

I think tabs in the settings menu are fine, but if we'd use tabs, I'd rather group the features (and fun) each as a tab and have the advanced settings as a collapsed region for every feature (otherwise this might be difficult to configure a single feature if you'd have to switch tabs all the time).

But we will see how this works best when we're testing v2 :-)

**Reply by jxn-30** [reply]
V2 introduces a tag system which allows adding multiple tags to a setting. This can be used to hide settings with specific tasks e.g. fun-settings.

Implementation is still very basic as it's still more a Proof-Of-Concept, however, I now prefer a tag system over collapsable sections / tabs as it doesn't nrcessarily add more types of interaction element to the settings modal.
